fs/span-manager/span_manager.go

```diff
@@ -304,52 +295,34 @@ func (m *SpanManager) getSpanInfo(offsetStart, offsetEnd compression.Offset) *sp
 	return &spanInfo
 }
 
-// GetSpanContent gets content (specified by start/end offset) from uncompressed
-// span data, which is returned as an `io.Reader`.
+// GetSpanContent gets uncompressed span content (specified by [offsetStart:offsetEnd]),
+// which is returned as an `io.Reader`.
+//
+// It resolves the span to ensure it exists and is uncompressed in cache:
+//  1. For `uncompressed` span, directly return the reader from the cache.
+//  2. For `fetched` span, read and uncompress the compressed span from cache, cache and
+//     return the reader from the uncompressed span.
+//  3. For `unrequested` span, fetch-uncompress-cache the span data, return the reader
+//     from the uncompressed span
+//  4. No span state lock will be acquired in `requested` state.
 func (m *SpanManager) GetSpanContent(spanID compression.SpanID, offsetStart, offsetEnd compression.Offset) (io.Reader, error) {
```

Comment:
Can we make `GetSpanContent` a private function?  Shouldn't every call be going through GetContents? Feels a bit odd that someone outside of the span manager would know the details of a span. I don't think this function is used anywhere other then in GetContents (which is used in fs/fs/reader.go).

Reply:
yeah agree. Updated `GetSpanContent` to private method